openhands/core/main.py

```diff
@@ -101,7 +103,17 @@ async def run_controller(
     if agent is None:
         agent = create_agent(runtime, config)
 
-    controller, initial_state = create_controller(agent, runtime, config)
+    replay_events: list[Event] | None = None
+    if config.replay_trajectory_path:
+        logger.info('Trajectory replay is enabled')
+        assert isinstance(initial_user_action, NullAction)
+        replay_events, initial_user_action = load_replay_log(
```

Comment:
nit: maybe we can assert to make sure the initial_user_action pass in this function to be `None` -- so that we don't sliently override the `initial_user_action` without the user knowing it

Reply:
See the line above it ^^

`L109: assert isinstance(initial_user_action, NullAction)`